### PR TITLE
implement a root check menu

### PR DESCRIPTION
**Pull request by nnyyxxxx**
## Type of Change
- [x] New feature

## Description
implements a root check menu

this is a better implementation of the now closed #821 
image

## Testing
image

### CODE REVIEW

File: tui/src/root.rs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                "\
+                WARNING: You are running this utility as root!\n\n\
+                This means you have full system access and commands can potentially damage your system if used incorrectly.\n\n\
+                Please proceed with caution and make sure you understand what each script does before executing it.\
+                ".into(),
+                "Root User Warning",
+                true,
+            );
+            let float: Float<dyn FloatContent> = Float::new(Box::new(warning), 60, 40);
+            state.set_float_window(float);
```
Comment (jeevithakannan2) [inline tui/src/root.rs:37]:
```suggestion
            state.focus = Focus::FloatingWindow(float);
```

File: tui/src/state.rs [hunk 2]
```diff
@@ -873,6 +873,10 @@ impl AppState {
         self.current_tab.select(Some(next));
         self.refresh_tab();
     }
+
+    pub fn set_float_window(&mut self, float: Float<dyn FloatContent + 'static>) {
+        self.focus = Focus::FloatingWindow(float);
+    }
```
Comment (jeevithakannan2) [inline tui/src/state.rs:879]:
```suggestion
```

Comment (nnyyxxxx) [reply tui/src/state.rs:879]: no, absolutely not.

File: tui/src/root.rs (hunk 1 above)
Comment (nnyyxxxx) [reply tui/src/root.rs:37]: no, absolutely not.

**Review by adamperkowski — APPROVED** [review]

**Comment by adamperkowski** [comment]
@ChrisTitusTech we need to fix the xtask workflow

**Comment by jeevithakannan2** [comment]
Can you elaborate how my 

> @ChrisTitusTech we need to fix the xtask workflow

No it's not broken. Someone didn't run xtask

**Comment by adamperkowski** [comment]
> > @ChrisTitusTech we need to fix the xtask workflow
> 
> No it's not broken. Someone didn't run xtask

Nope. The docs are now in a different repo. This workflow is useless.

**Comment by nnyyxxxx** [comment]
i dropped the docs changes

**Comment by nnyyxxxx** [comment]
@ChrisTitusTech ignore the ci

**Review by adamperkowski — APPROVED** [review]

**Review by lj3954 — APPROVED** [review]

**Review by jeevithakannan2 — APPROVED** [review]

**Review by ChrisTitusTech — APPROVED** [review]

**Pull request merged**